File: RSA/binary_exp.java
public static int binary_exp(int a, int e, int n){
    if(e == 0){
        return 1;
    }
    else if(e%2 == 0){
        return binary_exp(a, e/2, n) ** 2 % n;
    }
    else{
        return a* binary_exp(a,e-1,n) % n;
    }

}

File: RSA-Final/binary_exp.py
def binary_exp(a,e,n):
  if (e == 0):
    return 1
  elif (e%2 == 0):
    return binary_exp(a, e/2, n)** 2 % n
    # ((a**(e/2)) % n)**2 return binary_exp(a, e/2, n)**2
  else: 
    return a * binary_exp(a, e-1, n) % n
      # a * a**(e-1) % n